psl-core/src/main/java/org/linqs/psl/reasoner/sgd/SGDReasoner.java

```diff
@@ -62,6 +63,12 @@
 
     private float learningRate;
     private float learningRateInverseScaleExp;
+    private float epsilon;
+    private float adamBeta1;
+    private float adamBeta2;
+    float[] accumulatedGradientSquares;
+    float[] accumulatedGradientMean;
+    float[] accumulatedGradientVariance;
```

Comment:
Missing access modifiers.

But this point may be moot, since I'm not sure if these should be member variables.
They are allocated in optimize() and never deallocated, so they will just stick around for as long as the reasoner is around.
This could be potentially justified by saying that things like online inference or weight learning will call optimize() multiple times, but then this is still reallocated.
I'm also not a fan of how the terms call into the reasoner to get access to these arrays.
That makes for a circular coupling pattern (each one calls into the other) and is generally something to avoid.
(This circular coupling argument applies to the adamBetas as well.)

I think I have a good compromise:
- Make a static inner class (or actual class, your choice) that contains the three array references.
- Carry an object of this class around as a member variable.
- In optimize(), check to see if the proper array exists before allocating (note that type and size should be checked), you can also push that work off to the new class.